### PR TITLE
`VARC` table

### PR DESCRIPTION
**Pull request by behdad**
https://github.com/harfbuzz/boring-expansion-spec/blob/main/VARC.md

### CODE REVIEW

**Comment by behdad** [comment]
FontTools counterpart: https://github.com/fonttools/fonttools/pull/3395

**Comment by behdad** [comment]
This renders things now!

Need to clean up a bit: remove the old glyf1 varcomposites, add test, benchmark, etc.

**Comment by behdad** [comment]
> This renders things now!
> 
> Need to clean up a bit: remove the old glyf1 varcomposites, add test, benchmark, etc.

glyf1 varcomposites removed.

Just needs a test and some benchmarking.

**Comment by behdad** [comment]
Main TODO is subsetting closure over VARC, and subsetting VARC itself... Left for future.

**Comment by behdad** [comment]
Now with cycle detection, depth limit, and edge limit.

**Comment by behdad** [comment]
Both this and the fontttools PR are now up to date with the latest changes I made to the VARC table (including conditional components); and are have tests too. I think we can commit them. WDYT @rsheeter @anthrotype @khaledhosny

File: test/api/test-draw-varc.c [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+{
+  unsigned move_to_count;
+  unsigned line_to_count;
+  unsigned quad_to_count;
+  unsigned cubic_to_count;
+  unsigned close_path_count;
+} draw_data_t;
+
+/* Our modified itoa, why not using libc's? it is going to be used
+   in harfbuzzjs where libc isn't available */
```
Comment (khaledhosny) [inline test/api/test-draw-varc.c:40]: I don’t know what is the history of this comment, but we have libc when building harfbuzzjs for a while now. This also is a test file, so any harfbuzzjs concerns do not apply.

**Comment by khaledhosny** [comment]
This is way above my head, so I have nothing meaningful to add.

File: test/api/test-draw-varc.c (hunk 1 above)
Comment (behdad) [reply test/api/test-draw-varc.c:40]: Thanks. I just copy/pasted from `test-draw`. Let's leave.

**Comment by behdad** [comment]
> This is way above my head, so I have nothing meaningful to add.

Thanks for checking.

It should be straightforward translation of the spec:
https://github.com/harfbuzz/boring-expansion-spec/blob/main/VARC.md

**Pull request merged**